e2e/testcases/ssh_known_hosts_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	"kpt.dev/configsync/e2e/nomostest/policy"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	"kpt.dev/configsync/e2e/nomostest/testpredicates"
+	"kpt.dev/configsync/pkg/api/configsync"
+	"kpt.dev/configsync/pkg/core"
+	"kpt.dev/configsync/pkg/kinds"
+	"kpt.dev/configsync/pkg/reconcilermanager"
+	"kpt.dev/configsync/pkg/reconcilermanager/controllers"
+	"kpt.dev/configsync/pkg/status"
+)
+
+func TestRootReconcilerSSHKnownHost(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipNonLocalGitProvider)
+	var err error
+	rootSecret := &corev1.Secret{}
+	nt.T.Cleanup(func() {
+		nt.MustMergePatch(rootSecret, secretDataDeletePatch(controllers.KnownHostsKey))
+		nomostest.DeletePodByLabel(nt, "app", reconcilermanager.ManagerName, true)
```

Comment:
Why does the reconciler-manager Pod need to be deleted?

Reply:
from my observation, the root deployment isn't always updated after running `nt.MustMergePatch` in cleanup, which cause flakiness to the next running test.

Reply:
Hmm, I'm curious as to why that is. I expected the following events to happen:
- An update to the Secret triggers a watch event for reconciler-manager
- The Secret is mapped to the RootSync, triggering a reconcile
- During the reconcile loop, the known_hosts environment variable for the Deployment is set to false
- Changing the environment variable mutates the Deployment, which causes the Pod to be restarted

I wonder if one of these assumptions are incorrect.

Could you try replacing this with a wait to make ensure the Deployment is updated before proceeding?
```golang
	err = nt.Watcher.WatchObject(kinds.Deployment(),
		core.RootReconcilerPrefix, configsync.ControllerNamespace,
		[]testpredicates.Predicate{
			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "false"),
		},
	)
	if err != nil {
		nt.T.Error(err)
	}
```

Reply:
sorry looks like this was an oversight on my part. The controller in https://github.com/GoogleContainerTools/kpt-config-sync/pull/1001/files#diff-87f46982b7989d2ed226f9f1007d2327aae710a78e09c68bb4575ba18f0e340cR809 was previously only setting to true without any way of returning back to false (except by restarting the reconciler manager). Should be fixed now